File: 프로그래머스/1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.py
def solution(strings, n):
    answer = []
    dict = {}
    for i in strings:
        dict[i] = i[n]
    rank = sorted(dict.items(), key=lambda x: (x[1], x[0]))
    for i in rank:
        answer.append(i[0])
    return answer

File: 프로그래머스/1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.java
import java.util.*;


class Solution {
    public String[] solution(String[] strings, int n) {

        Arrays.sort(strings, (a, b) -> {
            if (a.charAt(n) != b.charAt(n)) {
                return Character.compare(a.charAt(n),b.charAt(n));
            } else {
                return a.compareTo(b);
            }
        });
        return strings;
    }
}